IntegrationTests/FolderModuleIntegrationSpecs.swift

```diff
@@ -0,0 +1,559 @@
+                    var folder: Folder?
+
+                    waitUntil(timeout: .seconds(Constants.defaultTimeout)) { done in
+                        self.client.folders.create(name: Folder.generateUniqueName(), parentId: "0") { result in
+                            switch result {
+                            case let .success(folderItem):
+                                folder = folderItem
+                            case let .failure(error):
+                                fail("Expected create call to succeed, but instead got \(error)")
+                            }
+
+                            done()
+                        }
+                    }
+
+                    guard let folder = folder else { return }
+
+                    waitUntil(timeout: .seconds(Constants.defaultTimeout)) { done in
+                        self.client.folders.delete(folderId: folder.id) { result in
+                            if case let .failure(error) = result {
+                                fail("Expected delete call to succeed, but instead got \(error)")
+                            }
+
+                            done()
+                        }
+                    }
+                }
+            }
+
+            context("get") {
+                var folderName: String!
+                var folder: Folder?
+
+                beforeEach {
+                    folderName = Folder.generateUniqueName()
+                    self.createFolder(name: folderName) { createdFolder in folder = createdFolder }
+                }
+
+                afterEach {
+                    self.deleteFolder(folder)
+                }
+
+                it("should correctly get a folder information") {
+                    guard let folder = folder else {
+                        fail("An error occurred during setup initial data")
+                        return
+                    }
+
+                    waitUntil(timeout: .seconds(Constants.defaultTimeout)) { done in
+                        self.client.folders.get(folderId: folder.id) { result in
+                            switch result {
+                            case let .success(folderItem):
+                                expect(folderItem).toNot(beNil())
+                                expect(folderItem).to(beAKindOf(Folder.self))
+                                expect(folderItem.name).to(equal(folderName))
+                                expect(folderItem.parent?.id).to(equal("0"))
+                                expect(folderItem.itemStatus).to(equal(.active))
+                            case let .failure(error):
+                                fail("Expected get call to suceeded, but instead got \(error)")
+                            }
+
+                            done()
+                        }
+                    }
+                }
+            }
+
+            context("list") {
+                var folder: Folder?
+
+                beforeEach {
```

Comment:
Are `beforeEach` and `afterEach` called before every `it` statement in this context? (I'm not that familiar with Quick) Was that your intention to recreate folder structure after line 147? Other tests are contained in a single `it`.

Reply:
`beforeEach` and `afterEach` are called before every it.
But these tests are focused on different approaches to fetching items so I put them in separate `it` blocks.